lib/semian/atomic_enum.rb

```diff
@@ -0,0 +1,40 @@
+module Semian
+  module AtomicEnumSharedImplementation
```

Comment:
conventional, this module should be called "AtomicEnum" or the filename should be "atomic_enum_shared_implementation.rb".